File: mgyoutube-multitech/api-node/src/webservices/modulereporegistry.ts
import { UserDataRepo } from '../repos/userdatarepo';
import { UserModule } from '../modules/usermodule';
import { DefaultUserModuleImpl } from '../modules/defaultusermoduleimpl';
import { VideoModule } from '../modules/videomodule';
import { SearchesModule } from '../modules/searchesmodule';


export class ModuleRepoRegistry {
    private static theUserModule: UserModule | null = null;
    private static theUserDataRepo: UserDataRepo | null = null;
    private static theVideoModule: VideoModule;
    private static theSearchesModule: SearchesModule;

    public static getUserModule(): UserModule {
        if (ModuleRepoRegistry.theUserModule == null) {
            var udr = ModuleRepoRegistry.getUserDataRepo()
            ModuleRepoRegistry.theUserModule = new DefaultUserModuleImpl(udr as UserDataRepo);
        }
        return ModuleRepoRegistry.theUserModule;
    }

    public static setUserModule(userModule: UserModule) {
        ModuleRepoRegistry.theUserModule = userModule;
    }

    public static getUserDataRepo(): UserDataRepo | null {
        return ModuleRepoRegistry.theUserDataRepo;
    }

    public static setUserDataRepo(userDataRepo: UserDataRepo) {
        ModuleRepoRegistry.theUserDataRepo = userDataRepo;

        if (ModuleRepoRegistry.theUserModule != null) {
            ModuleRepoRegistry.theUserModule.setUserDataRepo(userDataRepo);
        }
    }

    public static setVideosModule(videoModule: VideoModule) {
        ModuleRepoRegistry.theVideoModule = videoModule;
    }

    public static getVideosModule(): VideoModule {
        return ModuleRepoRegistry.theVideoModule;
    }

    public static setSearchesModule(searchesModule: SearchesModule) {
        ModuleRepoRegistry.theSearchesModule = searchesModule;
    }

    public static getSearchesModule(): SearchesModule {
        return ModuleRepoRegistry.theSearchesModule;
    }


}

File: mgyoutube-multitech/api-python/webservices/ModuleRepoRegistry.py
from modules import UserModule, DefaultUserModuleImpl, VideoModule, SearchesModule
from repos import UserDataRepo


class ModuleRepoRegistry(object):

    theUserModule: UserModule = None
    theUserDataRepo: UserDataRepo = None
    theVideoModule: VideoModule = None
    theSearchesModule: SearchesModule = None

    def __init__(self):
        pass

    @classmethod
    def getUserModule(cls) -> UserModule:
        if (cls.theUserModule is None):
            cls.theUserModule = DefaultUserModuleImpl.DefaultUserModuleImpl(
                cls.getUserDataRepo())
        return cls.theUserModule

    @classmethod
    def setUserModule(cls, userModule: UserModule):
        cls.theUserModule = userModule

    @classmethod
    def getUserDataRepo(cls) -> UserDataRepo:
        return cls.theUserDataRepo

    @classmethod
    def setUserDataRepo(cls, userDataRepo: UserDataRepo):
        cls.theUserDataRepo = userDataRepo

    @classmethod
    def getVideoModule(cls) -> VideoModule:
        return cls.theVideoModule

    @classmethod
    def setVideoModule(cls, videoModule: VideoModule):
        cls.theVideoModule = videoModule

    @classmethod
    def getSearchesModule(cls) -> SearchesModule:
        return cls.theSearchesModule

    @classmethod
    def setSearchesModule(cls, searchesModule: SearchesModule):
        cls.theSearchesModule = searchesModule
